src/main/scala/org/constellation/p2p/Download.scala

```diff
@@ -426,12 +412,23 @@ class DownloadProcess[F[_]: Concurrent: Timer: Clock](
       .flatMap(_ => snapshotsProcessor.processSnapshots(hashes))
       .flatTap(_ => dao.metrics.updateMetricAsync("downloadSecondPassComplete", "true"))
 
-  private def finishDownload(snapshot: SnapshotInfo): F[Unit] =
+  private def removeSnapshotInfoFromDisk(snapshotHeight: Long,
+                                         overWritePath: String = dao.snapshotInfoPath.pathAsString) = {
+    val snapshotInfoAtHeight = dao.snapshotInfoPath.collectChildren(_.isDirectory)
+      .filter { file =>
+        val Array(height, hash) = file.name.split('_')
+        height == snapshotHeight.toString
+      }.toList//need to call toList here
+    Sync[F].delay { snapshotInfoAtHeight.foreach(_.delete()) }
```

Comment:
That `F` **won't** wait for `delete` to finish when chaining so you basically did something like `unsafeRunSyncAndForget` inside IO.

So TL:DR: It should be `traverse` instead of `foreach` as thanks to that next `IO` will wait for delete to finish

Reply:
Exactly, that's some wild code